src/components/TopBar/ProjectPicker.jsx

```diff
@@ -21,22 +21,69 @@ const ProjectPicker = createMenu({
       isUserAuthenticated;
   },
 
-  renderItems({currentProjectKey, projectKeys, onChangeCurrentProject}) {
-    return map(projectKeys, projectKey => (
+  renderItems({
+    archivedViewOpen,
+    currentProjectKey,
+    projects,
+    onChangeCurrentProject,
+    onProjectUnArchived,
+    onToggleViewArchived,
+  }) {
+    const items = reduce(projects, (projectItems, project) => {
+      if (!project.archived) {
+        projectItems.push(
+          <MenuItem
+            isActive={project.projectKey === currentProjectKey}
+            key={project.projectKey}
+            onClick={partial(onChangeCurrentProject, project.projectKey)}
+          >
+            <ProjectPreview projectKey={project.projectKey} />
+          </MenuItem>,
+        );
+      } else if (archivedViewOpen && project.archived) {
+        projectItems.push(
+          <MenuItem
+            isActive={project.projectKey === currentProjectKey}
+            key={project.projectKey}
+            onClick={partial(onProjectUnArchived, project.projectKey)}
+          >
+            <ProjectPreview projectKey={project.projectKey} />
+          </MenuItem>,
+        );
+      }
+      return projectItems;
+    }, []);
+
+    items.push(
       <MenuItem
-        isActive={projectKey === currentProjectKey}
-        key={projectKey}
-        onClick={partial(onChangeCurrentProject, projectKey)}
+        isSticky
+        key="archivedProjectButton"
+        onClick={onToggleViewArchived}
       >
-        <ProjectPreview projectKey={projectKey} />
-      </MenuItem>
-    ));
+
+        {archivedViewOpen ?
+          <div>
+            Hide Archived Projects
+            <span className="u__icon"> &#xf077; </span>
+          </div> :
+          <div>
+            View Archived Projects
+            <span className="u__icon"> &#xf078; </span>
+          </div>
+        }
+      </MenuItem>,
+    );
+
+    return items;
   },
 })(ProjectPickerButton);
 
 ProjectPicker.propTypes = {
+  archivedViewOpen: PropTypes.bool.isRequired,
```

Comment:
The name suggests that there is a discrete archived view that may or may not be visible—I would call this something like `shouldShowArchivedProjects`

Reply:
Makes sense